FFMpegCore/FFMpeg/Pipes/RawAudioPipeSource.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace FFMpegCore.Pipes
+{
+    /// <summary>
+    /// Implementation of <see cref="IPipeSource"/> for a raw audio stream that is gathered from <see cref="IEnumerator{IAudioFrame}"/> 
+    /// </summary>
+    public class RawAudioPipeSource : IPipeSource
+    {
+        private readonly IEnumerator<IAudioSample> _sampleEnumerator;
+
+        public string Format { get; set; } = "s16le";
+        public uint SampleRate { get; set; } = 8000;
+        public uint Channels { get; set; } = 1;
+
+        public RawAudioPipeSource(IEnumerator<IAudioSample> sampleEnumerator)
+        {
+            _sampleEnumerator = sampleEnumerator;
+        }
+
+        public RawAudioPipeSource(IEnumerable<IAudioSample> sampleEnumerator)
+            : this(sampleEnumerator.GetEnumerator()) { }
+
+        public string GetStreamArguments()
+        {
+            return $"-f {Format} -ar {SampleRate} -ac {Channels}";
+        }
+
+        public async Task WriteAsync(Stream outputStream, CancellationToken cancellationToken)
+        {
+            if (_sampleEnumerator.Current != null)
```

Comment:
Is there a reason for not using a `foreach` loop?

Reply:
I took the `RawVideoPipeSource` as a direct reference, and `IEnumerator` can't be used with `foreach` statements directly

Reply:
Alright, makes sense